### PR TITLE
Custom Rubocop

### PR DESCRIPTION
**Pull request by Jeremy-Walton**
## Task

[Custom Rubocop Cop](https://github.com/RoleModel/airfield_designer/pull/47)

## Why?

To ensure consistent form error responses

## What Changed

* [X] Add custom cop to generator
* [X] Update docs to show all the current generators

### CODE REVIEW

File: lib/generators/rolemodel/linters/rubocop/templates/app/cops/form_error_response.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+# Node matcher (def_node_matcher :form_error)
+# This scans the block of code (in this case, the else branch of the if block)
+# using AST (Abstract Tree Syntax) to match certain patterns
+
+# Regex
+# This checks the specific matching line to ensure it meets our expectations
+
+# Custom Cop to Ensure Form Error Response Statuses
+class FormErrorResponse < RuboCop::Cop::Cop
```
Comment (KXEinc) [inline lib/generators/rolemodel/linters/rubocop/templates/app/cops/form_error_response.rb:16]: https://docs.rubocop.org/rubocop/v1_upgrade_notes.html#base-class

```suggestion
class FormErrorResponse < RuboCop::Cop::Base
```
RuboCop::Cop::Base has been deprecated:
Use RuboCop::Cop::Base instead.

File: lib/generators/rolemodel/linters/rubocop/README.md [hunk 2]
```diff
@@ -3,3 +3,4 @@
 ## What you get
 
 A Rubocop installation with a standard .rubocop.yml config.
+A Custom Cop that ensures form validation error responses use the correct status of unprocessable entity.
```
Comment (BenEinwechter) [inline lib/generators/rolemodel/linters/rubocop/README.md:6]:
```suggestion
A custom cop that ensures form validation error responses use the correct status of `422 Unprocessable Entity`.
```

File: lib/generators/rolemodel/linters/rubocop/rubocop_generator.rb [hunk 3]
```diff
@@ -1,7 +1,10 @@
+# frozen_string_literal: true
+
 require_relative '../../../bundler_helpers'
 
 module Rolemodel
   module Linters
+    # Install Standard rubocop and custom Cop
```
Comment (BenEinwechter) [inline lib/generators/rolemodel/linters/rubocop/rubocop_generator.rb:7]: Unneeded capitalization
```suggestion
    # Install the standard rubocop and a custom cop
```

**Review by KXEinc — CHANGES_REQUESTED** [review]
Just one change and this PR should be good to go.

**Review by BenEinwechter — APPROVED** [review]
A few small grammar suggestions, but looks good

**Review by KXEinc — APPROVED** [review]
LGTM

**Pull request merged**